lib/promise.rb

```diff
@@ -67,33 +81,67 @@ def rescue(&block)
   alias_method :catch, :rescue
 
   def sync
-    if pending?
-      wait
-      raise BrokenError if pending?
+    case @state
+    when :fulfilled
+      return @value
+    when :rejected
+      raise @reason
     end
-    raise reason if rejected?
-    value
+
+    wait
```

Comment:
This was not changed for the micr-benchmark, but mostly to make the code a bit clearer. If you think it was clearer before, I'm happy to change it back again! ❤️